File: web/src/types/classroom.ts
export interface Classroom {
	id: string;
	title: string;
	description: string;
	location: string;
	capacity: number;
	scheduled_at: string;
	classroom_code: string;
	created_at: string;
	updated_at: string;
}

export interface Participant {
	id: string;
	classroom_id: string;
	name: string;
	role: "learner" | "supporter" | "organizer";
	created_at: string;
}

File: api/internal/model/classroom.go
package model

import "time"

type Classroom struct {
	ID            string    `json:"id"`
	Title         string    `json:"title"`
	Description   string    `json:"description"`
	Location      string    `json:"location"`
	Capacity      int       `json:"capacity"`
	ScheduledAt   time.Time `json:"scheduled_at"`
	ClassroomCode string    `json:"classroom_code"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type Participant struct {
	ID          string    `json:"id"`
	ClassroomID string    `json:"classroom_id"`
	Name        string    `json:"name"`
	Role        string    `json:"role"`
	CreatedAt   time.Time `json:"created_at"`
}
